File: experiments/esbmc_limit_sweep/phase2_cases/06_dict/bad.c
#include <assert.h>

int nondet_int(void);

int lookup(int k)
{
    if (k == 1)
        return 10;

    if (k == 2)
        return 26;

    return 30;
}

int main(void)
{
    int k = nondet_int();

    __ESBMC_assume(1 <= k && k <= 3);

    int r = lookup(k);

    int expected;

    if (k == 1)
        expected = 10;
    else if (k == 2)
        expected = 25;
    else
        expected = 30;

    assert(r == expected);

    return 0;
}

File: experiments/esbmc_limit_sweep/phase2_cases/06_dict/bad.py
def lookup(k: int) -> int:
    table: dict[int, int] = {
        1: 10,
        2: 20,
        3: 30
    }

    # MUTATION
    table[2] = table[2] + 6

    return table[k]


k: int = nondet_int()

__ESBMC_assume(1 <= k <= 3)

r: int = lookup(k)

expected: int = 0

if k == 1:
    expected = 10
elif k == 2:
    expected = 25
else:
    expected = 30

assert r == expected
